server/game/gameSystems/DiscardCardsFromHandSystem.ts

```diff
@@ -63,16 +75,19 @@ export class DiscardCardsFromHandSystem<TContext extends AbilityContext = Abilit
         for (const player of properties.target as Player[]) {
             const availableHand = player.hand.filter((card) => properties.cardCondition(card, context));
 
+            // Select this player if its their own hand, or the active player from the context if its the 'opponent' choosing
+            const discardingPlayer = properties.discardingPlayerType === RelativePlayer.Self ? player : context.player;
+
             Contract.assertNonNegative(derive(properties.amount, player));
 
             const amount = Math.min(availableHand.length, derive(properties.amount, player));
 
-            if (amount === 0) {
+            if (amount === 0 && discardingPlayer.autoSingleTarget) {
                 events.push(this.generateEvent(context, additionalProperties));
                 continue;
             }
 
-            if (amount >= availableHand.length) {
+            if (amount >= availableHand.length && discardingPlayer.autoSingleTarget) {
```

Comment:
We need to consider the random case here, if it's random then we still shouldn't let them choose

Reply:
I believe this is handled.  If auto single target is disabled, it doesn't generate the events here,  The immediate next section of the code checks for *random* only preventing it from getting to the promptForSelect regardless of the auto setting